File: app/src/main/java/com/aite/a/activity/li/activity/BaseActivity.java
package com.aite.a.activity.li.activity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.aite.a.activity.li.util.LogUtils;
import com.jiananshop.a.R;

import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class BaseActivity extends AppCompatActivity implements View.OnClickListener{
    protected Context context;
    protected abstract int getLayoutResId();
    protected abstract void initView();
    protected abstract void initModel();
    private Unbinder unbinder;


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(getLayoutResId());
        context=this;
        unbinder= ButterKnife.bind(this);
        initView();
        initModel();
    }



    @Override
    public void onClick(View v) {

    }
    /**
     * 跳转页面
     *
     * @param clz 所跳转的目的Activity类
     */
    public void startActivity(Class<?> clz) {
        startActivity(new Intent(this, clz));
    }
    /**
     * 跳转页面
     *
     * @param clz 所跳转的目的Activity类
     */
    public void startActivity(Class<?> clz,String tag,String extra) {
        Intent intent=new Intent(this,clz);
        intent.putExtra(tag,extra);
        startActivity(intent);
    }

    /**
     * 跳转页面
     *
     * @param clz    所跳转的目的Activity类
     * @param bundle 跳转所携带的信息
     */
    public void startActivity(Class<?> clz, Bundle bundle) {
        Intent intent = new Intent(this, clz);
        if (bundle != null) {
            intent.putExtras(bundle);
        }
        startActivity(intent);
    }

    protected void initToolbar(String title) {
        try {
            ImageView backImg = this.findViewById(R.id.iv_back);
            backImg.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    onBackPressed();
                }
            });
            TextView titleTv = this.findViewById(R.id.tv_title);
            titleTv.setText(title);
        } catch (Exception e) {
            LogUtils.e("initToolbar-fail" + e);
        }
    }
    @Override
    protected void onDestroy() {
        super.onDestroy();
        unbinder.unbind();
    }
}

File: baselibrary/src/main/java/com/valy/baselibrary/basekotlin/BaseActivity.kt
package com.aite.mainlibrary.basekotlin

import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.annotation.StringRes
import androidx.appcompat.app.AppCompatActivity
import com.aite.awangdalibrary.utils.ScreenSizeUtils
import com.valy.baselibrary.R
import com.valy.baselibrary.utils.LogUtils
import com.valy.baselibrary.utils.StatusBarUtils
import me.goldze.mvvmhabit.utils.ToastUtils
import java.text.DecimalFormat

/**Lqayou

 * @Auther: liziyang

 * @datetime: 2020-01-18

 * @desc:
 * ?.意思是这个参数可以为空,并且程序继续运行下去

 * !!.的意思是这个参数如果为空,就抛出异常

 */
abstract class BaseActivity : AppCompatActivity() {
    abstract fun getLayoutId(): Int
    abstract fun initViews()
    abstract fun initDatas()
    protected var back: ImageView? = null
    protected var toolbar_title: TextView? = null
    protected lateinit var tv_right_title: TextView
    protected var mContext: Context? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayoutId())
        mContext = this
        initExtra()
        initViews()
        initDatas()
    }

    override fun onStart() {
        super.onStart()
        onClick()
    }

    /**
     * 为二次封装留下的初始化
     */
    open fun initExtra() {
    }

    /**
     * onclick
     */
    open fun onClick() {}

    /**
     * ->0默认为白色
     * ->1透明
     * ->string html5颜色
     */
    protected fun setStatusBar(s: Any) {
        when (s) {

            0 -> {
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                    StatusBarUtils.setColor(this, getColor(R.color.white))
                }
            }
            1 -> {
                StatusBarUtils.setTransparent(mContext)
            }

            else -> {
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                    StatusBarUtils.setColor(this, Color.parseColor(s.toString()))
                }
            }
        }
    }


    /**
     * 系统栏的高度
     */
    protected fun getStatusBarHeight(): Int {
        return StatusBarUtils.getHeight(mContext!!)
    }

    /**
     * app toolbar 初始化
     */
    protected fun initToolBar(title: CharSequence) {
        back = this.findViewById(R.id.iv_back)
        back?.setOnClickListener { onBackPressed() }
        toolbar_title = this.findViewById(R.id.tv_title)
        toolbar_title?.text = title


    }

    protected fun initToolBar(title: CharSequence, rightTitle: CharSequence) {
        back = this.findViewById(R.id.iv_back)
        back?.setOnClickListener { onBackPressed() }
        toolbar_title = this.findViewById(R.id.tv_title)
        tv_right_title = this.findViewById(R.id.tv_right_title)
        toolbar_title?.text = title
        tv_right_title?.text = rightTitle



    }

    open fun hideKeyboard(view: View) {
        val imm = getSystemService(
                Context.INPUT_METHOD_SERVICE) as InputMethodManager
        imm.hideSoftInputFromWindow(view.windowToken, 0)
    }


    /**
     * toast
     */
    protected fun showToast(msg: String) {
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show()

    }

    /**
     * 显示短时吐司
     *
     * @param resId 资源Id
     */
    open fun showShort(@StringRes resId: Int) {
        Toast.makeText(mContext, resId, Toast.LENGTH_SHORT).show()
    }

    /**
     * 跳转activity
     */
    fun startActivity(mClz: Class<*>) {
        startActivity(Intent(this, mClz))
    }

    fun startActivity(clz: Class<*>, bundle: Bundle?) {
        val intent = Intent(this, clz)
        if (bundle != null) {
            intent.putExtras(bundle)
        }
        startActivity(intent)
    }

    /**
     * 屏幕的尺寸
     */
    protected fun getScreenWidth(): Int {
        return ScreenSizeUtils.getScreenWidth(mContext!!)
    }

    protected fun getScreenHeight(): Int {
        return ScreenSizeUtils.getScreenHeight(mContext!!)
    }

    /**
     * 转换字符为小数后两位
     * 格式化，区小数后两位
     */
    protected open fun haveTwoDouble(d: Double): String? {
        val df = DecimalFormat("0.00")
        return try {
            df.format(d)
        } catch (e: Exception) {
            LogUtils.e(d)
            ""
        }
    }
}